mm2src/mm2_main/src/database/stats_nodes.rs

```diff
@@ -59,10 +68,17 @@ pub fn select_peers_addresses(ctx: &MmArc) -> SqlResult<Vec<(String, String)>, S
 }
 
 pub fn select_peers_names(ctx: &MmArc) -> SqlResult<HashMap<String, String>, SqlError> {
-    ctx.sqlite_connection()
+    #[cfg(not(feature = "new-db-arch"))]
+    let conn = ctx.sqlite_connection();
+    #[cfg(feature = "new-db-arch")]
+    let conn = ctx.global_db().unwrap();
+    // TODO: Can't use `conn` in the return statement because it's a mutex borrow, and also clippy complains when assigning the result into a temporary `result`.
+    #[allow(clippy::let_and_return)]
+    let result = conn
         .prepare(SELECT_PEERS_NAMES)?
         .query_map([], |row| Ok((row.get(0)?, row.get(1)?)))?
-        .collect::<SqlResult<HashMap<String, String>>>()
+        .collect::<SqlResult<HashMap<String, String>>>();
+    result
```

Comment:
in this case, we need to explicitly drop conn after we're done with it.

```rust
pub fn select_peers_names(ctx: &MmArc) -> SqlResult<HashMap<String, String>, SqlError> {
    #[cfg(not(feature = "new-db-arch"))]
    let conn = ctx.sqlite_connection();
    #[cfg(feature = "new-db-arch")]
    let conn = ctx.global_db().unwrap();
    let peers_names = conn
        .prepare(SELECT_PEERS_NAMES)?
        .query_map([], |row| Ok((row.get(0)?, row.get(1)?)))?
        .collect::<SqlResult<HashMap<String, String>>>();
    drop(conn);

    peers_names
}
```

Reply:
ummmm... no it should be dropped since peer_names is owned.

Reply:
well, compiler doesn't think so.

Reply:
i'm not really sure we need to drop here 😅
it's already dropped implicitly and compiles just fine.

Reply:
```rust
pub fn select_peers_names(ctx: &MmArc) -> SqlResult<HashMap<String, String>, SqlError> {
    #[cfg(not(feature = "new-db-arch"))]
    let conn = ctx.sqlite_connection();
    #[cfg(feature = "new-db-arch")]
    let conn = ctx.global_db();
    let mut stmt = conn.prepare(SELECT_PEERS_NAMES)?;
    let result = stmt
        .query_map([], |row| Ok((row.get(0)?, row.get(1)?)))?
        .collect::<SqlResult<HashMap<String, String>>>()?;

    Ok(result)
}
```

please check for unused imports.

Reply:
still pending..

```
// TODO: Can't use `conn` in the return statement because it's a mutex borrow, and also clippy complains when assigning the result into a temporary `result`.
    #[allow(clippy::let_and_return)]
```